### PR TITLE
update bep005 with latest master

### PR DESCRIPTION
**Pull request by sappelhoff**
- adds the latest master updates to the bep-005 branch
- importantly, this includes some BEP001 updates such as the entity linked file collections appendix, which is now available to link to
- due to the previous point, I re-introduce strict checking of links - all links should now pass again

I had to resolve a conflict in `src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md` in the `### Sequence Specifics` section, please have a careful look at that.

I'd be grateful if some of you could double check @patsycle @agahkarakuzu

### CODE REVIEW

**Comment by sappelhoff** [comment]
I used squash and merge, so if we identify an issue, we can simply revert that commit. https://github.com/bids-standard/bids-specification/commit/ef6f851d989e36ab8d0701a6225ac2b99b3f36a7

File: src/02-common-principles.md [hunk 1]
```diff
@@ -161,6 +161,28 @@ A summary of all entities in BIDS and the order in which they MUST be
...
+the parameter changes they declare MUST NOT invalidate the definition of the accompanying suffix.
+For example, the use of the `echo` entity along with the `T1w` suffix casts doubt on
+the validity of the identified contrast weighting.  
+Provided the conditions above are satisfied,
+any suffix (such as `bold`) can identify an entity-linked file collection,
+although certain suffixes are exclusive for this purpose (for example, `MP2RAGE`).
+Use cases concerning this convention are compiled in the
+[file collections](./99-appendices/10-file-collections.md) appendix.
+This convention is mainly intended for but not limited to MRI modalities.
+
```
Comment (guiomar) [inline src/02-common-principles.md:185]: Could a practical example be added so it's easier to understand?

Comment (sappelhoff) [reply src/02-common-principles.md:185]: cc @agahkarakuzu

Comment (agahkarakuzu) [reply src/02-common-principles.md:185]: @guiomar we have example datasets at https://osf.io/k4bs5/. However, we need to update those file names to account for the the latest changes. We have a few PRs left to complete BEP001, which will make this clearer.

**Pull request merged**